bundles/org.openhab.binding.modbus.kermi/src/main/java/org/openhab/binding/modbus/kermi/internal/handler/KermiXcenterThingHandler.java

```diff
@@ -0,0 +1,666 @@
+                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, ALARM_REG_START, ALARM_REG_SIZE, 3);
+                alarmPoller = localComms.registerRegularPoll(alarmRequest, STATE_POLL_REFRESH_TIME_MS, 0,
+                        this::handleAlarmResult, this::handleAlarmFailure);
+
+                pollTasks.add(alarmPoller);
+
+                ModbusReadRequestBlueprint stateRequest = new ModbusReadRequestBlueprint(slaveId,
+                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, STATE_REG_START, STATE_REG_SIZE, 3);
+                statePoller = localComms.registerRegularPoll(stateRequest, STATE_POLL_REFRESH_TIME_MS, 0,
+                        this::handleStateResult, this::handleStateFailure);
+
+                pollTasks.add(statePoller);
+
+                // default polling speed
+                ModbusReadRequestBlueprint chargingcircuitRequest = new ModbusReadRequestBlueprint(slaveId,
+                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, CHARGING_CIRCUIT_REG_START,
+                        CHARGING_CIRCUIT_REG_SIZE, 3);
+                chargingCircuitPoller = localComms.registerRegularPoll(chargingcircuitRequest, localConfig.refresh, 0,
+                        this::handleChargingCircuitResult, this::handleChargingCircuitFailure);
+
+                pollTasks.add(chargingCircuitPoller);
+
+                ModbusReadRequestBlueprint energySourceRequest = new ModbusReadRequestBlueprint(slaveId,
+                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, ENERGY_SOURCE_REG_START, ENERGY_SOURCE_REG_SIZE,
+                        3);
+                energySourcePoller = localComms.registerRegularPoll(energySourceRequest, localConfig.refresh, 0,
+                        this::handleEnergySourceResult, this::handleEnergySourceFailure);
+
+                pollTasks.add(energySourcePoller);
+
+                ModbusReadRequestBlueprint powerRequest = new ModbusReadRequestBlueprint(slaveId,
+                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, POWER_REG_START, POWER_REG_SIZE, 3);
+                powerPoller = localComms.registerRegularPoll(powerRequest, localConfig.refresh, 0,
+                        this::handlePowerResult, this::handlePowerFailure);
+
+                pollTasks.add(powerPoller);
+
+                if (localConfig.pvEnabled) {
+                    ModbusReadRequestBlueprint pvRequest = new ModbusReadRequestBlueprint(slaveId,
+                            ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, PV_MODULATION_REG_START,
+                            PV_MODULATION_REG_SIZE, 3);
+                    pvPoller = localComms.registerRegularPoll(pvRequest, localConfig.refresh, 0, this::handlePvResult,
+                            this::handlePvFailure);
+
+                    pollTasks.add(pvPoller);
+                }
+
+            } // else state handling performed in connectEndPoint function
```

Comment:
This is not true, the `connectEndPoint` immediate returns null of it is null. so that case is not handled and the Thign will always remain in the UNKOWN state. 
I think there should be a task rescheduled to try again, or atleast put the Thing into Offline state.